R/rplot.R

```diff
@@ -0,0 +1,94 @@
+#' @param y_range Vector of type numeric. The range of \code{y}; it can be used to clip the image. For larger datasets, providing \code{y_range}
+#' may result in improved performance.
+#' @param background Character. The background color of the image to plot.
+#' @param reduction_func Function. A reduction function is used to aggregate data points into their pixel representations. Currently
+#' supported reduction operators are \code{sum}, \code{any}, \code{mean}, \code{m2}, \code{first}, \code{last}, \code{min} and \code{max}. Default is \code{sum}. See details.
+#' @param layout Character. The method used to generate layouts for multiple images. The default is \code{weighted}. Useful for categorical
+#' data (i.e. "color" is provided via \code{aes()}). \code{weighted} specifies that the final raster should be a weighted combination of each
+#' (categorical) aggregation matrix. Conversely, \code{cover} indicates that the afterwards objects will be drawn on top of 
+#' the previous ones.
+#' @param glyph Character. Currently, only "circle" and "square" are supported; as the \code{size} of the pixels increases, how should they
+#' spread out -- should the pattern be circular or square? Other glyphs may be added in the future.
+#' 
+#' @export
+#' @examples 
+#' if(requireNamespace("ggplot2")) {
+#'   library(ggplot2)
+#'   # `color` represents a variable here
+#'   with(diamonds, 
+#'        rplot(x = carat, y = price, color = color)
+#'   )
+#'   # `color` represents an actual color vector
+#'   with(diamonds, 
+#'        rplot(x = carat, y = price, color = fire_map)
+#'   )
+#' }
+rplot.default <- function(x, y = NULL, ..., 
+                          plot_width = 600, plot_height = 600,
+                          x_range = NULL, y_range = NULL,
+                          background = "white",
+                          reduction_func = NULL,
+                          layout = NULL,
+                          glyph = NULL) {
+  
+  xy <- grDevices::xy.coords(x, y)
+  data <- data.table::data.table(x = xy$x, y = xy$y)
+  mapping <- aes(x = x, y = y)
+  n <- dim(data)[1]
+  
+  args <- list(...)
+  
+  for(aes_name in setdiff(rasterly_aesthetics, c("x", "y"))) {
+    var <- args[[aes_name]]
+
+    if(length(var) == n) {
+      data <- data[, rlang::expr_name(aes_name) := var]
+      aes_names <- names(mapping)
+      mapping <- mbind(rlang::quo(!!rlang::sym(aes_name)), mapping) %>% 
+        stats::setNames(., c(aes_name, aes_names))
+      args[[aes_name]] <- NULL
+    }
+  }
+
+  r <- do.call(
+    rasterly,
+    c(
+      list(
+        data = data,
+        mapping = mapping,
+        plot_width = plot_width,
+        plot_height = plot_height,
+        background = background,
+        x_range = x_range,
+        y_range = y_range
+      ),
+      args
+    )
+  )  %>% 
+    rasterly_points(reduction_func = reduction_func,
+                    layout = layout,
+                    glyph = glyph) %>% 
+    rasterly_build()
+  
+  print(r)
```

Comment:
May want to remove this if not necessary.

Reply:
Check this 
```
# nothing is printed
library(rasterly)
library(ggplot2)
for(i in 1:4) {
  rasterly(diamonds, 
           mapping = aes(x = carat, y = price), 
           size = i) %>% 
    rasterly_points()
}
```

```
for(i in 1:4) {
  rasterly(diamonds, 
           mapping = aes(x = carat, y = price), 
           size = i) %>% 
    rasterly_points() %>% 
    print()
}
```
When the expression is wrapped, `print` will not be executed by default. So, call `print` can force the graphics to be created.